src/EnergyPlus/ElectricPowerServiceManager.cc

```diff
@@ -2026,6 +2051,14 @@ namespace EnergyPlus {
 			thermalPowerOutput = thermProdRate;
 			break;
 		}
+		case GeneratorType::pvWatts: {
+			PVWatts::PVWattsGenerator &pvwattsGenerator(PVWatts::GetOrCreatePVWattsGenerator(name));
```

Comment:
This name isn't passed around every simulation call right?  Kinda looks like it is, and if so, we need to change that to have this object keep a reference to the correct generator and just call it each time.  We won't look up the object every single call.  

If this is a one-time thing and I'm just reading it wrong, then nothing to change.

Reply:
No, we're passing the name of the generator and looking it up each time. In my defense, look at the other `Sim*Generator` calls on the preceding and following lines. They're all doing the same thing, except redundantly passing both the index and the name. I was trying to make my class fit into that pattern.

Reply:
Yeah, I get it, it's a bit of a balancing act, between continuing E+ patterns and breaking them to make the whole thing better.  I'm not going to make you change it for this.  In the future, keeping a reference to an object and only looking it up once then directly operating on the reference is my preferred path forward.